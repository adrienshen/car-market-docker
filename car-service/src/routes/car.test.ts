import request from 'supertest';
import server from '../index';

test('GET: /v1/cars/:id works', async done => {
    return request(server).get('/v1/cars/JHk290Xj')
        .then(response => {
            expect(response.status).toBe(200);
            expect(response.text).toMatchSnapshot();
            expect(response.body.id).toBe('JHk290Xj');
            done();
        })
        .catch(err => done(err))
});

test('GET: /v1/cars works', async done => {
    return request(server).get('/v1/cars')
        .then(response => {
            expect(response.status).toBe(200);
            expect(response.text).toMatchSnapshot();
            expect(response.body.length).toBe(4);
            done();
        })
        .catch(err => done(err));
});

test('GET: v1/cars?make=Toyota&model=Rav4 works', async done => {
    return request(server).get('/v1/cars?make=Toyota&model=Rav4')
        .then(response => {
            expect(response.status).toBe(200);
            expect(response.body.length).toBe(1);
            expect(response.text).toMatchSnapshot();
            done();
        })
        .catch(err => done(err));
});

test('GET: v1/cars?make=Toyota&model=Rav4?mileage_lt=9000 should not return results', async done => {
    return request(server).get('/v1/cars?make=Toyota&model=Rav4&mileage_lt=9000')
        .then(response => {
            expect(response.status).toBe(404);
            expect(response.body.length).toBe(0);
            done();
        })
        .catch(err => done(err));
});
