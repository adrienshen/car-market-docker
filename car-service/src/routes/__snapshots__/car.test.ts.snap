// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET: /v1/cars works 1`] = `"[{\\"id\\":\\"JHk290Xj\\",\\"make\\":\\"Ford\\",\\"model\\":\\"F10\\",\\"package\\":\\"Base\\",\\"color\\":\\"Silver\\",\\"year\\":2010,\\"category\\":\\"Truck\\",\\"mileage\\":120123,\\"price_cents\\":1999900,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"},{\\"id\\":\\"fWI37la\\",\\"make\\":\\"Toyota\\",\\"model\\":\\"Camry\\",\\"package\\":\\"SE\\",\\"color\\":\\"White\\",\\"year\\":2019,\\"category\\":\\"Sedan\\",\\"mileage\\":3999,\\"price_cents\\":2899000,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"},{\\"id\\":\\"1i3xjRllc\\",\\"make\\":\\"Toyota\\",\\"model\\":\\"Rav4\\",\\"package\\":\\"XSE\\",\\"color\\":\\"Red\\",\\"year\\":2018,\\"category\\":\\"SUV\\",\\"mileage\\":24001,\\"price_cents\\":2275000,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"},{\\"id\\":\\"dku43920s\\",\\"make\\":\\"Ford\\",\\"model\\":\\"Bronco\\",\\"package\\":\\"Badlands\\",\\"color\\":\\"Burnt Orange\\",\\"year\\":2022,\\"category\\":\\"SUV\\",\\"mileage\\":1,\\"price_cents\\":4499000,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"}]"`;

exports[`GET: /v1/cars/:id works 1`] = `"{\\"id\\":\\"JHk290Xj\\",\\"make\\":\\"Ford\\",\\"model\\":\\"F10\\",\\"package\\":\\"Base\\",\\"color\\":\\"Silver\\",\\"year\\":2010,\\"category\\":\\"Truck\\",\\"mileage\\":120123,\\"price_cents\\":1999900,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"}"`;

exports[`GET: v1/cars?make=Toyota&model=Rav4 works 1`] = `"[{\\"id\\":\\"1i3xjRllc\\",\\"make\\":\\"Toyota\\",\\"model\\":\\"Rav4\\",\\"package\\":\\"XSE\\",\\"color\\":\\"Red\\",\\"year\\":2018,\\"category\\":\\"SUV\\",\\"mileage\\":24001,\\"price_cents\\":2275000,\\"created_at\\":\\"2021-06-20T18:58:46.170Z\\",\\"updated_at\\":\\"2021-06-20T18:58:46.170Z\\"}]"`;
